x/expiration/client/cli/tx.go

```diff
@@ -0,0 +1,138 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+
+	"github.com/provenance-io/provenance/x/expiration/types"
+)
+
+const (
+	FlagSigners = "signers"
+)
+
+// NewTxCmd is the top-level command for expiration CLI transactions
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Aliases:                    []string{"exp"},
+		Short:                      "Transaction commands for the expiration module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	txCmd.AddCommand(
+		ExtendExpirationCmd(),
+		InvokeExpirationCmd(),
+	)
+	return txCmd
+}
+
+// ExtendExpirationCmd creates a command for extending an expiration
+func ExtendExpirationCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "extend [module-asset-id] [duration n{y|w|d|h}]",
+		Aliases: []string{"e"},
+		Short:   "Extend expiration metadata for an asset on the provenance blockchain",
+		Long: `Extend expiration metadata for an asset on the provenance blockchain
+
+Example:
+$ %s tx expiration extend pb1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk duration 1y
+
+module-asset-id   - module asset address
+duration          - the duration period for which the module asset will continue to remain on chain.
+					Valid time units are "y", "w", "d", "h"`,
+		Example: fmt.Sprintf(`$ %s tx expiration extend pb1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk duration 1y`, version.AppName),
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			signers, err := parseSigners(cmd, &clientCtx)
+			if err != nil {
+				return err
+			}
+
+			moduleAssetID := args[0]
+			duration := args[1]
+			msg := types.NewMsgExtendExpirationRequest(moduleAssetID, duration, signers)
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	addSignerFlagCmd(cmd)
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// InvokeExpirationCmd creates a command for invoking expiration logic on an asset
+func InvokeExpirationCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "invoke [module-asset-id]",
+		Aliases: []string{"i"},
+		Short:   "Invoke expiration logic for an asset on the provenance blockchain",
```

Comment:
I haven't looked at the code for Invoking yet, but one question that I have as a new user is "will invoke logic be called again when the timer expires?". Might be nice to clarify that either here or in the specs.

Reply:
>  "will invoke logic be called again when the timer expires?". Might be nice to clarify that either here or in the specs.

Invocation is performed by a user. Will specify that in the specs.